### PR TITLE
Migrate from XCTest to Swift Testing

### PR DESCRIPTION
**Pull request by mattt**
This PR migrates most of our tests from XCTest to the new Swift Testing framework, which has better ergonomics and performance.

### CODE REVIEW

File: Tests/GenerationTests/TestUtils.swift [hunk 1]
```diff
@@ -1,22 +1,7 @@
...
-        let rhsEvaluated = try expression2()
-        XCTAssertEqual(lhsEvaluated.count, rhsEvaluated.count, file: file, line: line)
-        for (lhs, rhs) in zip(lhsEvaluated, rhsEvaluated) {
-            XCTAssertEqual(lhs, rhs, accuracy: accuracy, file: file, line: line)
-        }
-    } catch {
-        XCTFail("Unexpected error: \(error)", file: file, line: line)
-    }
+/// Check if two floating-point arrays are equal within a given accuracy
+func isClose<T: FloatingPoint>(_ lhs: [T], _ rhs: [T], accuracy: T) -> Bool {
```
Comment (pcuenca) [inline Tests/GenerationTests/TestUtils.swift:4]: Nice. This could become a general utility function later, like Pytorch or MLX `allClose`.

**Comment by pcuenca** [comment]
> Are parallel tokenizer tests still revealing race conditions?

Ah, I see they work now!

File: Tests/GenerationTests/TestUtils.swift (hunk 1 above)
Comment (mattt) [reply Tests/GenerationTests/TestUtils.swift:4]: Yeah, that would be nice. Official guidance from the Swift team for replacing the XCTest assertion with accuracy is to pull in swift-numerics to use their implementation of [`isApproximatelyEqual`](https://github.com/apple/swift-numerics/blob/be0d7c65d1321e0cddee84acba10daf5f13ff795/Sources/RealModule/ApproximateEquality.swift#L72). Pulling that in as a separate dependency just for tests is overkill, but we could certainly vendor it.

**Comment by mattt** [comment]
> > Are parallel tokenizer tests still revealing race conditions?
> 
> Ah, I see they work now!

Yes, that's the race conditions I was mentioning before. There's a more comprehensive change we can do to make that reentrancy safe, but for now it's much easier to ensure that we don't have more than one test calling `AutoTokenizer.from(pretrained:)` with the same model.

**Comment by mattt** [comment]
Ooh, nice — after making everything parallel again, we're now down to 3 minutes 40 seconds 🚀
